File: src/main/java/me/anton/sickcore/bootstrap/VelocityBootstrap.java
package me.anton.sickcore.bootstrap;

import com.google.inject.Inject;
import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent;
import com.velocitypowered.api.event.proxy.ProxyShutdownEvent;
import com.velocitypowered.api.plugin.Plugin;
import com.velocitypowered.api.proxy.ProxyServer;
import lombok.Getter;
import me.anton.sickcore.core.ProxyCore;
import org.slf4j.Logger;

@Getter
@Plugin(id = "sickcore", name = "SickCore", version = "1.0.0-SNAPSHOT", description = "Core for SickNetwork", authors = "Anton")
public class VelocityBootstrap {

    @Getter
    private final ProxyServer server;
    @Getter
    private final Logger logger;
    private ProxyCore core;

    @Inject
    public VelocityBootstrap(ProxyServer server, Logger logger) {
        this.server = server;
        this.logger = logger;
    }

    @Subscribe
    public void onProxyLoad(ProxyInitializeEvent event){
        core = new ProxyCore(this);
        core.onLoad();
    }

    @Subscribe
    public void onProxyUnLoad(ProxyShutdownEvent event){
        core.onUnLoad();
    }
}

File: src/main/java/me/anton/sickcore/core/VelocityBootstrap.kt
package me.anton.sickcore.core

import com.google.inject.Inject
import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent
import com.velocitypowered.api.event.proxy.ProxyShutdownEvent
import com.velocitypowered.api.plugin.Plugin
import com.velocitypowered.api.proxy.ProxyServer
import org.slf4j.Logger

@Plugin(
    id = "SickCore_Velocity",
    name = "SickCore",
    version = "1.1",
    description = "Velocity Core of SickNetwork",
    url = "discord.sickmc.net",
    authors = ["btwonion"]
)

class VelocityBootstrap @Inject constructor(val server: ProxyServer,val logger: Logger) {

    private var core: VelocityCore? = null

    @Subscribe
    suspend fun onProxyInitialize(event: ProxyInitializeEvent){
        core = VelocityCore(this)
        core!!.start()
    }

    @Subscribe
    suspend fun onProxyShutdown(event: ProxyShutdownEvent){
        core!!.shutdown()
    }


}